procrastinate/testing.py

```diff
@@ -192,17 +192,24 @@ def finished_jobs(self) -> list[JobRow]:
     def fetch_job_one(self, queues: Iterable[str] | None) -> dict:
         # Creating a copy of the iterable so that we can modify it while we iterate
 
-        for job in self.jobs.values():
+        filtered_jobs = [
+            job
+            for job in self.jobs.values()
             if (
                 job["status"] == "todo"
                 and (queues is None or job["queue_name"] in queues)
                 and (not job["scheduled_at"] or job["scheduled_at"] <= utils.utcnow())
                 and job["lock"] not in self.current_locks
-            ):
-                job["status"] = "doing"
-                self.events[job["id"]].append({"type": "started", "at": utils.utcnow()})
+            )
+        ]
+
+        filtered_jobs.sort(key=lambda job: job["priority"], reverse=True)
```

Comment:
Hm, I haven't seen `.sort()` in years, I guess it's more usual to used `sorted` that doesn't work by side effect.

Also (unrelated):
I know that Python's sort is stable but for the sake of clarity, we should sort by id (ASC) too. Theas means: `key=lambda job: (-job["priority"], job["id"])` (and no `reverse)`

Reply:
> Hm, I haven't seen `.sort()` in years, I guess it's more usual to used `sorted` that doesn't work by side effect.

Yeah, I am more of an older guy (quite an old profile picture 😉).

> Also (unrelated): I know that Python's sort is stable but for the sake of clarity, we should sort by id (ASC) too. Theas means: `key=lambda job: (-job["priority"], job["id"])` (and no `reverse)`

That's why I used `sort` instead of `sorted` initially, as I also sorted by `id` (and I could have it in one lambda). Then I thought it was not necessary, as it is initially sorted by `id`. But, of course, let's make it explicit.